core/src/main/java/com/github/kaktushose/jda/commands/embeds/Embed.java

```diff
@@ -0,0 +1,317 @@
+        this.placeholders.putAll(Arrays.stream(placeholders).collect(Collectors.toUnmodifiableMap(I18n.Entry::name, I18n.Entry::value)));
+        return this;
+    }
+
+    /// Returns a [MessageEmbed] just like [EmbedBuilder#build()], but will also localize this embed based on the
+    /// [#locale(Locale)] and [`placeholders`][#placeholder(String, Object)] provided.
+    ///
+    /// @return the built, sendable [MessageEmbed]
+    @NotNull
+    @Override
+    public MessageEmbed build() {
+        String json = super.build().toData().toString();
+        try {
+            JsonNode node = localize(mapper.readTree(json));
+            return EmbedBuilder.fromData(DataObject.fromJson(node.toString())).build();
+        } catch (JsonProcessingException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    private JsonNode localize(JsonNode node) {
+        if (node instanceof ObjectNode objectNode) {
+            Iterator<Map.Entry<String, JsonNode>> iterator = objectNode.fields();
+            while (iterator.hasNext()) {
+                Map.Entry<String, JsonNode> entry = iterator.next();
+                JsonNode child = entry.getValue();
+                JsonNode newChild = localize(child);
+                if (newChild.isTextual()) {
+                    objectNode.put(entry.getKey(), i18n.localize(locale, newChild.asText(), placeholders));
+                } else {
+                    objectNode.set(entry.getKey(), newChild);
+                }
+            }
+        } else if (node instanceof ArrayNode arrayNode) {
+            for (int i = 0; i < arrayNode.size(); i++) {
+                JsonNode child = arrayNode.get(i);
+                arrayNode.set(i, localize(child));
+            }
+        }
+        return node;
+    }
+
+    /// Transforms this embed into [MessageCreateData].
+    @NotNull
+    public MessageCreateData toMessageCreateData() {
+        return MessageCreateData.fromEmbeds(build());
+    }
+
+    /// Transforms this embed into [MessageEditData].
+    @NotNull
+    public MessageEditData toMessageEditData() {
+        return MessageEditData.fromEmbeds(build());
+    }
+
+    /// Methods for manipulating the fields of an [Embed].
+    public interface Fields {
+
+        /// Removes all fields of this embed based on the given [Predicate].
+        ///
+        /// @param filter the [Predicate] to test the fields with
+        /// @return this instance for fluent interface
+        @NotNull
+        Fields removeIf(@NotNull Predicate<MessageEmbed.Field> filter);
+
+        /// Removes all fields with the given name of this embed based on the given [Predicate].
+        ///
+        /// @param name   the name of the fields to test
+        /// @param filter the [Predicate] to test the fields with
+        /// @return this instance for fluent interface
+        @NotNull
+        Fields removeIf(@NotNull String name, Predicate<MessageEmbed.Field> filter);
```

Comment:
Can be implemented as a default method by calling `removeIf(Predicate)` with the same logic as above in the implementation.